CMakeLists.txt

```
@@ -0,0 +1,96 @@
+# Minimum cmake version required by scikit-build-core: 3.15
+# Minimum cmake version required by `Python_SOABI`: 3.17
+cmake_minimum_required(VERSION 3.17)
+
+set(CLP_FFI_PY_INSTALL_LIBS ON CACHE BOOL "Whether to install the built libraries." FORCE)
+
+if (DEFINED SKBUILD_PROJECT_NAME)
+    set(CLP_FFI_PY ${SKBUILD_PROJECT_NAME} CACHE STRING
+            "Use scikit-build-core's project name." FORCE)
+    set(CLP_FFI_PY_VERSION ${SKBUILD_PROJECT_VERSION} CACHE STRING
+            "Use scikit-build-core's project version." FORCE)
+else()
+    # The project is configured outside of scikit-build-core. Normally this is for build testing
+    # purpose, or loading the project into CLion. We don't install the built library in this config.
+    message(WARNING "Configured outside of scikit-build-core. Library installation is disabled.")
+    set(CLP_FFI_PY_INSTALL_LIBS OFF CACHE BOOL
+            "Disable library installation when building outside of scikit-build-core." FORCE)
+    set(CLP_FFI_PY "clp-ffi-py-local-build" CACHE STRING "Use local build name." FORCE)
+    set(CLP_FFI_PY_VERSION "0.0.12" CACHE STRING "Use local build version." FORCE)
+    # Enable compile commands by default if the generator supports it.
+    if (NOT CMAKE_EXPORT_COMPILE_COMMANDS AND CMAKE_GENERATOR MATCHES "Ninja|Unix Makefiles")
+        set(CMAKE_EXPORT_COMPILE_COMMANDS ON CACHE BOOL
+                "Enable/Disable output of compile commands during generation." FORCE)
+    endif()
+endif()
+
+project(${CLP_FFI_PY}
+    LANGUAGES CXX C
+    VERSION ${CLP_FFI_PY_VERSION}
+)
+
+# C-extension modules will be compiled as shared libraries. To enable dynamic linking, all libraries
+# (including the static ones) must be compiled with position-independent-code option.
+set(CMAKE_POSITION_INDEPENDENT_CODE ON CACHE BOOL "Compile libraries using PIC.")
+
+# Find Python interpreter, compiler and development environment.
+find_package(Python COMPONENTS Interpreter Development.Module REQUIRED)

```

Shall we place the `REQUIRED` keyword before specifying the components?

```suggestion
find_package(Python REQUIRED COMPONENTS Interpreter Development.Module)
```